components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/core/management/PreUpdatePasswordActionDTOModelResolver.java

```diff
@@ -0,0 +1,274 @@
+     * When properties are not updated, the existing properties should be sent to the upstream component.
+     *
+     * @param updatingActionDTO  ActionDTO that needs to be updated.
+     * @param existingActionDTO  Existing ActionDTO.
+     * @param tenantDomain       Tenant domain.
+     * @return Resolved ActionDTO.
+     * @throws ActionDTOModelResolverException ActionDTOModelResolverException.
+     */
+    @Override
+    public ActionDTO resolveForUpdateOperation(ActionDTO updatingActionDTO, ActionDTO existingActionDTO,
+                                               String tenantDomain) throws ActionDTOModelResolverException {
+
+        Map<String, Object> properties = new HashMap<>();
+        resolveCertificateUpdate(updatingActionDTO, existingActionDTO, properties, tenantDomain);
+        resolvePasswordSharingFormatUpdate(updatingActionDTO, existingActionDTO, properties);
+
+        return new ActionDTO.Builder(updatingActionDTO)
+                .properties(properties)
+                .build();
+    }
+
+    @Override
+    public void resolveForDeleteOperation(ActionDTO deletingActionDTO, String tenantDomain)
+            throws ActionDTOModelResolverException {
+
+        if (deletingActionDTO.getProperty(CERTIFICATE) instanceof Certificate) {
+            Certificate certificate = (Certificate) deletingActionDTO.getProperty(CERTIFICATE);
+            deleteCertificate(certificate.getId(), tenantDomain);
+        }
+    }
+
+    private static Certificate buildCertificate(String actionId, Certificate updatingCertificate) {
+
+        return new Certificate.Builder()
+                .name("ACTIONS:" + actionId)
+                .certificateContent(updatingCertificate.getCertificateContent())
+                .build();
+    }
+
+    private void resolvePasswordSharingFormatUpdate(ActionDTO updatingActionDTO, ActionDTO existingActionDTO,
+                                                    Map<String, Object> properties) {
+
+        if (updatingActionDTO.getProperty(PASSWORD_SHARING_FORMAT) != null) {
+            properties.put(PASSWORD_SHARING_FORMAT,
+                    ((PasswordSharing.Format) updatingActionDTO.getProperty(PASSWORD_SHARING_FORMAT)).name());
+        } else {
+            properties.put(PASSWORD_SHARING_FORMAT,
+                    ((PasswordSharing.Format) existingActionDTO.getProperty(PASSWORD_SHARING_FORMAT)).name());
+        }
+    }
+
+    private void resolveCertificateUpdate(ActionDTO updatingActionDTO, ActionDTO existingActionDTO,
+                                          Map<String, Object> properties, String tenantDomain)
+            throws ActionDTOModelResolverException {
+
+        Certificate updatingCertificate = (Certificate) updatingActionDTO.getProperty(CERTIFICATE);
+        Certificate existingCertificate = (Certificate) existingActionDTO.getProperty(CERTIFICATE);
+
+        if (isNewCertificateAdding(updatingCertificate, existingCertificate)) {
```

Comment:
isAddingNewCertificate
isDeletingExistingCertificate
isUpdatingExistingCertificate

Better term like above so it's easy to read

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6281/commits/604c0addbbf444856b2f38e31bf645ccf07965b4